vdist/configuration.py

```diff
@@ -0,0 +1,117 @@
+from __future__ import absolute_import
+
+import string
+import sys
+# The ConfigParser module has been renamed to configparser in Python 3
+if sys.version_info[0] == 3:
+    import configparser
+else:
+    import ConfigParser as configparser
+
+import vdist.source as source
+
+LISTABLE_ARGUMENTS = {"source_git", "source_git_directory", "runtime_deps",
+                      "build_deps"}
+LONG_TEXT_ARGUMENTS = {"fpm_args", "pip_args"}
+PROCESSABLE_ARGUMENTS = {"output_folder", "source_directory", "compile_python",
+                         "fpm_args"}
+USELESS_ARGUMENTS = {"mode"}
```

Comment:
`mode` is a console launcher argument. As you can read in [howtouse.md](https://github.com/objectified/vdist/pull/10/files#diff-9f6855a6ed566899313808a59cb789c2):

> Once you have a configuration file you can launch vdist from your console using
> **batch mode**:
> 
> ```bash
> $ vdist batch configuration_file
> ```
> 
> When launched vdist will create sequentially all packages configured in your
> file.
> 
> Batch mode is the usual mode your are going to use through console but vdist
> offers a **manual mode** too. That mode does not use a configuration file but
> allows you to set parameters as command arguments:
> 
> ```bash
> $ vdist manual --app geolocate --version 1.3.0 --source_git https://github.com/dante-signal31/geolocate,master --profile ubuntu-trusty --compile_python --python_version 3.4.4 --fpm_args '--maintainer dante.signal31@gmail.com -a native --url https://github.com/dante-signal31/geolocate --description "This program accepts any text and searchs inside every IP address." --license BSD-3 --category net' --requirements_path /REQUIREMENTS.txt --runtime_deps libssl1.0.0 dummy1.0.0 --output_folder ./dist
> ```
> 
> Pay attention to the point that `--fpm_args` argument is enclosed in single quotes.
> 
> Manual mode may be useful to dinamically set parameters through console scripts.

`mode`is the argument used by console launcher to know if it should read a configuration file (**batch mode**) or arguments passed through console launcher (**manual mode**), but once those parameters are read from their source `mode` argument is no longer useful (at least from `Configuration` instance point of view).